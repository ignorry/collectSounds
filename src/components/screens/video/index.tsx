import React, { lazy } from "react";
import ScreenWrapper from "../../../lib/screenWrapper";
import { useParams } from "react-router-dom";

import Spinner from "../../primitives/Spinner";

const Component = lazy( () => import( "./Component" ) );

const Search = () => {
  const params = useParams();

  return (
    <ScreenWrapper file="video" skeleton={ <Spinner/> }>
      <Component id={ params.id || null }/>
    </ScreenWrapper>
  );
}

export default Search;