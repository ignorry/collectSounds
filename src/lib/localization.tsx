import React, { ReactNode, useState, useEffect } from "react";
import { IntlProvider } from "react-intl";

export type Lang = 'en' | 'ru';
export type Page = 
  'search' |
  'saved' |
  'queue' |
  'settings' |
  'video' |
  'playlist' |
  'channel' |
  'synchronization';

/**
 * import specific translation file
 * @param {Page} page 
 * @param {Lang} lang 
 */
export const loadMessages = ( page: Page, lang: Lang ) => import( `../translations/${page}/${lang}.json` );

/**
 * get user's language
 * @returns {Lang} lang code (eg 'en', 'ru')
 */
export const getLang = (): Lang => {
  const code = navigator.language.slice( 0, 2 );

  if ( code === 'ru' ) return code;

  return 'en';
}

/**
 * @typedef Props
 * @prop {string} file - translation file
 * @prop {ReactNode} children
 */
export type Props = {
  file: Page;
  children?: ReactNode;
  skeleton?: ReactNode;
}

/**
 * wrap component with TranslationProvider
 * @param {Props} props
 */
export const LangProvider: React.FC<Props> = ( props: Props ) => {
  const locale = getLang();
  const [messages, setMessages] = useState( null );

  useEffect(() => {
    loadMessages( props.file, locale ).then( setMessages );
  }, [locale]);

  return messages ? (
    <IntlProvider locale={locale} messages={messages}>
      { props.children }
    </IntlProvider>
  ) : <> { props.skeleton } </>;
}