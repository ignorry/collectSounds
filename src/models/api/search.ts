import { PageInfo, Thumbnails } from "./primitives"

/**
 * @typedef SearchItem
 * @prop {string} kind - identifies the API resource's type. The value will be youtube#searchResult.
 * @prop {string} etag - the Etag of this resource.
 * @prop {object} id - contains information that can be used to uniquely identify the resource that matches the search request.
 * @prop {string} id.kind - the type of the API resource.
 * @prop {string} [id.videoId] - if video
 * @prop {string} [id.channelId] - if channel
 * @prop {string} [id.playlistId] - if playlist
 * @prop {object} [snippet] - contains basic details about the channel, such as its title, description, and thumbnail images.
 * @prop {string} [snippet.title]
 * @prop {string} [snippet.channelId]
 * @prop {string} [snippet.description]
 * @prop {string} [snippet.publishedAt]
 * @prop {Thumbnails} [snippet.thumbnails]
 * @prop {string} [snippet.channelTitle]
 * @prop {string} [snippet.liveBroadcastContent] - an indication of whether a video or channel resource has live broadcast content. Valid property values are upcoming, live, and none.
 */
export type SearchItem = {
  kind: string;
  etag: string;
  id: {
    kind: string;
    videoId?: string;
    channelId?: string;
    playlistId?: string;
  },
  snippet: {
    publishedAt: string;
    channelId: string;
    title: string;
    description: string;
    thumbnails: Thumbnails;
    channelTitle: string;
    liveBroadcastContent: string;
  }
}

/**
 * @typedef SearchResponse
 * @prop {string} kind - identifies the API resource's type. The value will be youtube#searchListResponse.
 * @prop {string} etag - the Etag of this resource.
 * @prop {string} nextPageToken - can be used as the value of the pageToken parameter to retrieve the next page in the result set.
 * @prop {string} prevPageToken - can be used as the value of the pageToken parameter to retrieve the previous page in the result set.
 * @prop {string} regionCode - the region code that was used for the search query.
 * @prop {Array<SearchItem>} [items] - a list of results that match the request criteria.
 * @prop {PageInfo} pageInfo - encapsulates paging information for the result set.
 */
export type SearchResponse = {
  kind: string;
  etag: string;
  nextPageToken: string;
  prevPageToken: string;
  regionCode: string;
  pageInfo: PageInfo;
  items: Array<SearchItem>;
}