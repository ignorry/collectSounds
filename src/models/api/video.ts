import { PageInfo, Localized, Thumbnails } from "./primitives";

/**
 * @typedef Channel
 * @prop {string} kind - identifies the API resource's type. The value will be youtube#channel.
 * @prop {string} etag - the Etag of this resource.
 * @prop {string} id
 * @prop {object} [snippet] - contains basic details about the channel, such as its title, description, and thumbnail images.
 * @prop {string} [snippet.title]
 * @prop {string} [snippet.description]
 * @prop {string} [snippet.publishedAt]
 * @prop {Thumbnails} [snippet.thumbnails]
 * @prop {Localized} [snippet.localized] - contains a localized title and description for the channel or it contains the channel's title and description in the default language for the channel's metadata.
 * @prop {string} [snippet.channelId]
 * @prop {string} [snippet.channelTitle]
 * @prop {Array<string>} [snippet.tags]
 * @prop {string} [snippet.categoryId]
 * @prop {string} [snippet.liveBroadcastContent]
 * @prop {string} [snippet.defaultLanguage]
 * @prop {string} [snippet.defaultAudioLanguage]
 * @prop {object} [contentDetails] - contains information about the video content, including the length of the video and an indication of whether captions are available for the video.
 * @prop {string} [contentDetails.duration] - the length of the video.
 * @prop {string} [contentDetails.dimension] - indicates whether the video is available in 3D or in 2D.
 * @prop {string} [contentDetails.definition] - indicates whether the video is available in high definition (HD) or only in standard definition.
 * @prop {string} [contentDetails.caption] - indicates whether captions are available for the video.
 * @prop {boolean} [contentDetails.licensedContent] - indicates whether the video represents licensed content, which means that the content was uploaded to a channel linked to a YouTube content partner and then claimed by that partner.
 * @prop {string} [contentDetails.contentRating] - pecifies the ratings that the video received under various rating schemes.
 * @prop {string} [contentDetails.projection] - specifies the projection format of the video. (360 | rectangular)
 * @prop {object} [statistics] - encapsulates statistics for the channel.
 * @prop {string} [statistics.viewCount]
 * @prop {string} [statistics.likeCount] 
 * @prop {string} [statistics.favoriteCount]
 * @prop {string} [commentCount]
 */
export type Video = {
  kind: string;
  etag: string;
  id: string;
  snippet?: {
    publishedAt: string;
    channelId: string;
    title: string;
    description: string;
    thumbnails: Thumbnails;
    channelTitle: string;
    tags: Array<string>;
    categoryId: string;
    liveBroadcastContent: string;
    defaultLanguage: string;
    localized: Localized;
    defaultAudioLanguage: string;
  },
  contentDetails?: {
    duration: string;
    dimension: string;
    definition: string;
    caption: string;
    licensedContent: boolean;
    contentRating: object;
    projection: string;
  };
  statistics?: {
    viewCount: string;
    likeCount: string;
    favoriteCount: string;
    commentCount: string;
  };
}

/**
 * @typedef VideoItem
 * @prop {string} kind - identifies the API resource's type. The value will be youtube#playlistItem.
 * @prop {string} etag - the Etag of this resource.
 * @prop {string} id
 */
export type VideoItem = {
  kind: string;
  etag: string;
  id: string;
}

/**
 * @typedef VideoIdResponse
 * @prop {string} kind - identifies the API resource's type. The value will be youtube#videoListResponse.
 * @prop {string} etag - the Etag of this resource.
 * @prop {Array<Video>} [items] - a list of videos that match the request criteria.
 * @prop {PageInfo} pageInfo - encapsulates paging information for the result set.
 */
export type VideoIdResponse = {
  kind: string;
  etag: string;
  items?: Array<Video>;
  pageInfo: PageInfo;
}

/**
 * @typedef PlaylistItemsResponse
 * @prop {string} kind - identifies the API resource's type. The value will be youtube#playlistItemListResponse.
 * @prop {string} etag - the Etag of this resource.
 * @prop {Array<VideoItem>} [items] - a list of videos in a playlist.
 * @prop {PageInfo} pageInfo - encapsulates paging information for the result set.
 */
export type PlaylistItemsResponse = {
  kind: string;
  etag: string;
  items?: Array<VideoItem>;
  pageInfo: PageInfo;
}