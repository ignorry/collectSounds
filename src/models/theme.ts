export enum ThemeEnum  {
  light = 'light',
  dark = 'dark'
}

// all sizes in rem
export type Theme = {
  iconSize: number,
  rangeSize: number,
  spinnerSize: number,
  contentWidth: number,
  colors: {
    main: string,
    bg?: string,
    bgSecondary?: string,
    font?: string,
    darkBg?: string,
  },
  decorative: { // in px
    borderRadius: number;
    padding: number;
  },
  gaps: {
    smallest: number,
    small: number,
    big: number,
  },
  thumbnail: {
    width: number,
    height: number,
    infoColor: string,
    infoWidth: 4,
  },
  error: {
    backgroundColor: string,
    width: number,
  },
  media: {
    small: string,
  },
  font: {
    primary: {
      size: number,
      opacity: number,
    },
    secondary: {
      size: number,
      opacity: number,
    },
  },
  durations: {
    ms300: number,
  },
  order: {
    footer: number,
    modal: number,
  },
}