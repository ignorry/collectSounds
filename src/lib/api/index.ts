import { 
  fetchVideo,
  fetchPlaylist,
  fetchPlaylistItems,
  fetchChannel,
  fetchSearch
} from "./fetch";
import {
  parseVideo,
  parsePlaylist,
  parseChannel
} from "./parseApiResponse";

import { Video, Playlist, Channel, Content } from "../../models/content";

/**
 * fetch video by id
 * @param {string} id - video id
 * @returns {Promise<Array<Video>>} 
 */
export const getVideoById = async ( id: string ): Promise<Array<Video>> => {
  const res = await fetchVideo({
    part: 'snippet,contentDetails',
    id
  });

  return parseVideo( res );
}

/**
 * fetch playlist by id
 * @param {string} id - playlist id
 * @returns {Promise<Array<Playlist>>}
 */
export const getPlaylistById = async ( id: string ): Promise<Array<Playlist>> => {
  const res = await fetchPlaylist({
    part: 'snippet',
    id
  });

  return parsePlaylist( res );
}

/**
 * fetch channel by id
 * @param {string} id - channel id
 * @returns {Promise<Array<Channel>>}
 */
export const getChannelById = async ( id: string ): Promise<Array<Channel>> => {
  const res = await fetchChannel({
    part: 'snippet',
    id
  });

  return parseChannel( res );
}

/**
 * fetch playlist's videos
 * @param {string} id - playlist id
 * @returns {Promise<Array<Video>>}
 */
export const getPlaylistVideos = async ( id: string ): Promise<Array<Video>> => {
  const res = await fetchPlaylistItems({
    part: 'id',
    playlistId: id,
  });

  const ids: Array<string> = res.items.map( item => item.id );

  return getVideoById( ids.join( ',' ) );
}

/**
 * fetch channel's videos
 * @param {string} id - channel id
 * @returns {Promise<Array<Video>>}
 */
export const getChannelVideos = async ( id: string ): Promise<Array<Video>> => {
  const res = await fetchSearch({
    part: 'id',
    channelId: id,
    order: 'date',
  })

  const ids: Array<string> = res.items.filter( item => item.id.kind === 'video' ).map( item => item.id.videoId );

  return getVideoById( ids.join( ',' ) );
}

/**
 * fetch channel's playlists
 * @param {string} id - channel id
 * @returns {Promise<Array<Playlist>>}
 */
export const getChannelPlaylists = async ( id: string ): Promise<Array<Playlist>> => {
  const res = await fetchPlaylist({
    part: 'snippet',
    channelId: id,
  });

  return parsePlaylist( res )
}

/**
 * @typedef SearchFilters
 * @prop {string} order - specifies the method that will be used to order resources in the API response
 * @prop {string} type - restricts a search query to only retrieve a particular type of resource
 * @prop {string} videoDuration - filters video search results based on their duration
 * @prop {string} saveSearch - indicates whether the search results should include restricted content as well as standard content
 */
export type SearchFilters = {
  order?: 'date' | 'rating' | 'relevance' | 'title' | 'videoCount' | 'viewCount';
  type?: 'video' | 'playlist' | 'channel',
  videoDuration?: 'any' | 'long' | 'medium' | 'short';
  saveSearch?: 'moderate' | 'none' | 'strict'
}

/**
 * fetch search results with specified filters
 * @param {SearchFilters} filters
 * @returns {Promise<Array<Content>>}
 */
export const getSearch = async ( filters: SearchFilters ): Promise<Array<Content>> => {
  const res = await fetchSearch({...filters,
    part: 'id',
    maxResults: 10,
  });

  const videoIds: Array<string> = [];
  const playlistIds: Array<string> = [];
  const channelIds: Array<string> = [];
  const ids: Array<string> = res.items.map( item => {
    if ( item.id.kind === 'video' ) {
      videoIds.push( item.id.videoId );
      return item.id.videoId;
    } else if ( item.id.kind === 'playlist' ) {
      playlistIds.push( item.id.playlistId );
      return item.id.playlistId;
    }

    channelIds.push( item.id.channelId );
    return item.id.channelId;
  });

  const videos: Array<Video> = await getVideoById( videoIds.join( ',' ) );
  const playlists: Array<Playlist> = await getPlaylistById( playlistIds.join( ',' ) );
  const channels: Array<Channel> = await  getChannelById( channelIds.join( ',' ) );

  // ids array needed to save order of contents

  return ids.map( item => {
    if ( videos[0] && videos[0].id === item ) return videos.shift();
    if ( playlists[0] && playlists[0].id === item ) return playlists.shift();
    return channels.shift();
  });
}