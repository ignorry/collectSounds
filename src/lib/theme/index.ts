import { DefaultTheme } from "styled-components";
import { ThemeEnum } from "../../models/theme";
import { baseTheme } from "./baseTheme";

export const lightTheme: DefaultTheme = {
  ...baseTheme,
  type: ThemeEnum.light,

  colors: {
    ...baseTheme.colors,
    main: '#00bb88',
    bg: '#ffffff',
    bgSecondary: '#d8d8d8',
    font: '#000000',
    darkBg: '#d9d9d9'
  },
}

export const darkTheme: DefaultTheme = {
  ...baseTheme,
  type: ThemeEnum.dark,

  colors: {
    ...baseTheme.colors,
    bg: '#000000',
    bgSecondary: '#2a2a2a',
    font: '#ffffff',
    darkBg: '#121212'
  },
}

/**
 * get theme depending on isDark parameter
 * @param {boolean} isDark 
 * @returns {DefaultTheme}
 */
export const getTheme = ( isDark: boolean ): DefaultTheme => isDark ? darkTheme : lightTheme;