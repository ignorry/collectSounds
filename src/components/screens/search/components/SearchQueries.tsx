import React, { useState } from "react";
import styled from "styled-components";
import { useIntl } from "react-intl";
import { SearchFilters } from "../../../../lib/api";

import Input from "../../../primitives/Input";
import Collapse from "../../../primitives/Collapse";
import Dropdown from "../../../primitives/Dropdown";
import Button from "../../../primitives/Button";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${ ({ theme }) => `${ theme.gaps.big }rem` };
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
  margin: auto;
  background: ${ ({ theme }) => theme.colors.bgSecondary };
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
`;

const InputWrapper = styled.div`
  display: flex;
  gap: ${ ({ theme }) => `${ theme.gaps.big }rem` };

  input {
    width: 100%;
  }
`;

/**
 * @typedef Props
 * @prop {Function} callback - calls with object of filters
 * @prop {Object} queries - some object of queries, used on first render
 */
export type Props = {
  callback: Function;
  queries: any;
}

const SearchQueries: React.FC<Props> = ( props: Props ) => {
  const intl = useIntl();
  const [q, setQ] = useState<SearchFilters["q"] | null>( props.queries?.q || null );
  const [order, setOrder] = useState<SearchFilters["order"]>( props.queries?.order || 'relevance' );
  const [type, setType] = useState<SearchFilters["type"] | 'all'>( props.queries?.type || 'all' );
  const [duration, setDuration] = useState<SearchFilters["videoDuration"] | 'any'>( props.queries?.videoDuration || 'any' );

  const handleSearch = () => {
    const filters: SearchFilters = {
      order: order,
    };
    if ( q ) filters.q = q;
    if ( type && type !== 'all' ) filters.type = type;
    if ( duration && duration !== 'any' ) filters.videoDuration = duration;

    props.callback( filters );
  };

  return (
    <Container>
      <InputWrapper>
        <Input value={ q } callback={ setQ } placeholder={ intl.formatMessage({ id: 'input' }) }/>
        <Button callback={ handleSearch } pic="search" active/>
      </InputWrapper>
      <Collapse label={ intl.formatMessage({ id: 'filters' })} icon="filters" dontHandleOutside>
        <Dropdown 
          callback={ setOrder }
          activeVal={ intl.formatMessage({ id: order }) } 
          values={[
            intl.formatMessage({ id: 'date' }),
            intl.formatMessage({ id: 'rating' }),
            intl.formatMessage({ id: 'relevance' }),
            intl.formatMessage({ id: 'title' }),
            intl.formatMessage({ id: 'videoCount' }),
            intl.formatMessage({ id: 'viewCount' }),
          ]}
        />
        <Dropdown 
          callback={ setType } 
          activeVal={ intl.formatMessage({ id: type }) } 
          values={[
            intl.formatMessage({ id: 'all' }),
            intl.formatMessage({ id: 'video' }),
            intl.formatMessage({ id: 'playlist' }),
            intl.formatMessage({ id: 'channel' }),
          ]}
        />
        <Dropdown 
          callback={ setDuration}
          activeVal={ intl.formatMessage({ id: duration }) } 
          values={[
            intl.formatMessage({ id: 'any' }),
            intl.formatMessage({ id: 'long' }),
            intl.formatMessage({ id: 'medium' }),
            intl.formatMessage({ id: 'short' }),
          ]}
        />
      </Collapse>
    </Container>
  )
};

export default SearchQueries;