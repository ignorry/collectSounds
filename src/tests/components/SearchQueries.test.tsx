import React from "react";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { getTheme } from "../../lib/theme";
import GlobalStyles from "../../lib/styled/global";
import { Provider } from "react-redux";
import store from "../../redux/index";
import { LangProvider } from "../../lib/localization";
import { BrowserRouter } from "react-router-dom";
import { SearchResponse } from "../../models/api/search";
import { VideoIdResponse } from "../../models/api/video";

import SearchQueries from "../../components/screens/search/components/SearchQueries";

const AllTheProviders = ( { children }: any ) => (
  <Provider store={store}>
    <ThemeProvider theme={ getTheme( true ) } >
      <BrowserRouter>
        <LangProvider file="root">
          { children }
        </LangProvider>
      </BrowserRouter>
      <GlobalStyles/>
    </ThemeProvider>
  </Provider>
);

describe( 'SearchQueries Component', () => {
  it( 'renders correctly', async () => {
    const result = render(
      <SearchQueries queries={{}} callback={() => {}}/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    const button = result.container.querySelector( 'button' );

    expect( button ).toBeTruthy();
  });

  it( 'handles click on search button', async () => {
    const callback = jest.fn();

    const result = render(
      <SearchQueries queries={{}} callback={callback}/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    result.container.querySelector( 'button' ).click();

    expect( callback ).toBeCalled();
  });

  it( 'handles click on search button with specified filters', async () => {
    const callback = jest.fn();

    const result = render(
      <SearchQueries queries={{
        q: 'q',
        type: 'video',
        videoDuration: 'short'
      }} callback={callback}/>,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    result.container.querySelector( 'button' ).click();

    expect( callback ).toBeCalled();
  });

  it( 'matches to snapshot', async () => {
    const result = render(
      <SearchQueries queries={{}} callback={() => {}} />,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 10 ) );

    expect( result ).toMatchSnapshot();
  });
});