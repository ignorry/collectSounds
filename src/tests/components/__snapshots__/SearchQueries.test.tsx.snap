// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchQueries Component matches to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-papXJ kEIprk"
      >
        <div
          class="sc-jqUVSM hWOGQF"
        >
          <input
            class="sc-bczRLJ dldCsv"
            placeholder="input"
            value=""
          />
          <button
            class="sc-gKXOVf feeGsc"
          >
            <svg
              class="sc-gsnTZi dRHeag"
            >
              <use
                href="http://localhost/icons.svg#search"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-eCYdqJ FxPOW"
        >
          <button
            class="sc-hKMtZM fntRFW"
          >
            <svg
              class="sc-gsnTZi gSmFuM"
            >
              <use
                href="http://localhost/icons.svg#filters"
              />
            </svg>
            <p
              class="sc-dkzDqf krpKaR"
            >
              filters
            </p>
          </button>
          <div
            class="sc-jSMfEi jmXqhj"
          >
            <div
              class="sc-iBkjds envIss"
            >
              <button
                class="sc-gKXOVf feeGsc"
              >
                <p
                  class="sc-dkzDqf QiTYm"
                >
                  relevance
                </p>
              </button>
              <ul
                class="sc-ftvSup hiVoSg"
              >
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      date
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      rating
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      title
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      videoCount
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      viewCount
                    </p>
                  </button>
                </li>
              </ul>
            </div>
            <div
              class="sc-iBkjds envIss"
            >
              <button
                class="sc-gKXOVf feeGsc"
              >
                <p
                  class="sc-dkzDqf QiTYm"
                >
                  all
                </p>
              </button>
              <ul
                class="sc-ftvSup hiVoSg"
              >
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      video
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      playlist
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      channel
                    </p>
                  </button>
                </li>
              </ul>
            </div>
            <div
              class="sc-iBkjds envIss"
            >
              <button
                class="sc-gKXOVf feeGsc"
              >
                <p
                  class="sc-dkzDqf QiTYm"
                >
                  any
                </p>
              </button>
              <ul
                class="sc-ftvSup hiVoSg"
              >
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      long
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      medium
                    </p>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf QiTYm"
                    >
                      short
                    </p>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-papXJ kEIprk"
    >
      <div
        class="sc-jqUVSM hWOGQF"
      >
        <input
          class="sc-bczRLJ dldCsv"
          placeholder="input"
          value=""
        />
        <button
          class="sc-gKXOVf feeGsc"
        >
          <svg
            class="sc-gsnTZi dRHeag"
          >
            <use
              href="http://localhost/icons.svg#search"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-eCYdqJ FxPOW"
      >
        <button
          class="sc-hKMtZM fntRFW"
        >
          <svg
            class="sc-gsnTZi gSmFuM"
          >
            <use
              href="http://localhost/icons.svg#filters"
            />
          </svg>
          <p
            class="sc-dkzDqf krpKaR"
          >
            filters
          </p>
        </button>
        <div
          class="sc-jSMfEi jmXqhj"
        >
          <div
            class="sc-iBkjds envIss"
          >
            <button
              class="sc-gKXOVf feeGsc"
            >
              <p
                class="sc-dkzDqf QiTYm"
              >
                relevance
              </p>
            </button>
            <ul
              class="sc-ftvSup hiVoSg"
            >
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    date
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    rating
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    title
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    videoCount
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    viewCount
                  </p>
                </button>
              </li>
            </ul>
          </div>
          <div
            class="sc-iBkjds envIss"
          >
            <button
              class="sc-gKXOVf feeGsc"
            >
              <p
                class="sc-dkzDqf QiTYm"
              >
                all
              </p>
            </button>
            <ul
              class="sc-ftvSup hiVoSg"
            >
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    video
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    playlist
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    channel
                  </p>
                </button>
              </li>
            </ul>
          </div>
          <div
            class="sc-iBkjds envIss"
          >
            <button
              class="sc-gKXOVf feeGsc"
            >
              <p
                class="sc-dkzDqf QiTYm"
              >
                any
              </p>
            </button>
            <ul
              class="sc-ftvSup hiVoSg"
            >
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    long
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    medium
                  </p>
                </button>
              </li>
              <li>
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf QiTYm"
                  >
                    short
                  </p>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
